### ISSUE TITLE
After computer shutting down unexpectedly I lost my saved tabs :(

### ISSUE THREAD

**Issue by georgiosd** [cxw42/TabFern#180]
Is there any way to recover them???

**Comment by cxw42** [comment]
Sorry to hear that!  There may be.  First of all, have you made any backups (menu | Backup now)?  If so, you can easily get back to your last backed-up state.

If not, would you please give me as much detail as you can?  OS, browser, TF versions; what happened; what you observe now; what have you tried; any error messages in the browser console (Ctrl+Shift+I in the TF window); any popup errors TF shows.

Please don't include any private information since GH issues are open to the public.  Thanks!  I'll take a look at what you provide and see if I have any ideas.

**Comment by georgiosd** [comment]
No backups. I wrongly assumed that Tabfern would keep some sort of internal backup...

Windows 10, Chrome latest, TF 0.2.0. 

Nothing "happened". Woke up and computer was frozen so had to shut it down. Rebooted, Tabfern was empty :o

No errors in devtools for TF. Tried restoring tabs from Chrome...

**Comment by georgiosd** [comment]
Oh and, the "Local Extension Settings" directory has some .ldb files of about 3.5 MB each and a "lost" directory.

**Comment by georgiosd** [comment]
Looks like tabs are saved in .log files but the one with last modified = today has data only about the middle and then zeros... Which must have happened when the computer shut down.

**Comment by cxw42** [comment]
Automatic backup is an open request but not yet implemented, unfortunately.  This tells me I at least need to update the docs to be more clear on that point!

Whatever you do, please first copy your profile directory as it presently stands, including the`lost` directory.  I will think about whether I can add a function that will attempt to recover data.

I looked online for Chrome-extension data recovery and found several suggestions to use the Windows "Previous Versions" feature to get back to an earlier profile directory.  I think that's worth a try.

**Comment by georgiosd** [comment]
@cxw42 should be fairly easy to write the data to disk once a day or week right? You dont even need to add options for it, no one is going to complain about having some safety for a few megabytes of storage. 

I did check the Previous Versions but since I dont use System Restore that was a lost cause too.

I ended up extracting as many URLs as I could from the half-corrupted file and then dowloading my browsing history from Google, and extracting the URLs from there too. It will take a while to comb through them but I'll know I've got everything :)

Thanks for trying.
